tests/test_notify_suppliers_with_incomplete_applications.py

```diff
@@ -0,0 +1,166 @@
+from logging import Logger
+import mock
+import pytest
+
+from dmutils.email import DMNotifyClient
+from dmscripts.notify_suppliers_with_incomplete_applications import (
+    notify_suppliers_with_incomplete_applications,
+    MESSAGES,
+)
+
+
+FRAMEWORK_SUPPLIERS_TEST_CASES = [
+    [
+        # input
+        [{
+            'supplierId': 0,
+            'applicationCompanyDetailsConfirmed': True,
+            'declaration': {'status': 'complete', 'primaryContactEmail': 'abc@example.com'}
+        }],
+        ['abc@example.com', ],  # output, list of email addresses
+        '',  # output, expected message fragment
+    ],
+    [
+        # input
+        [{
+            'supplierId': 1,
+            'applicationCompanyDetailsConfirmed': True,
+            'declaration': {'status': 'started', 'primaryContactEmail': 'abc@example.com'}
+        }],
+        ['abc@example.com', ],  # output, list of email addresses
+        MESSAGES[1],  # output, expected message fragment
+    ],
+    [
+        # input
+        [{
+            'supplierId': 2,
+            'applicationCompanyDetailsConfirmed': True,
+            'declaration': {'status': 'started'}
+        }],
+        [],  # output, list of email addresses
+        MESSAGES[1],  # output, expected message fragment
+    ],
+    [
+        # input
+        [{
+            'supplierId': 3,
+            'applicationCompanyDetailsConfirmed': True,
+            'declaration': {}
+        }],
+        [],  # output, list of email addresses
+        MESSAGES[1],  # output, expected message fragment
+    ],
+    [
+        # input
+        [{
+            'supplierId': 4,
+            'applicationCompanyDetailsConfirmed': None,
+            'declaration': {'status': 'complete', 'primaryContactEmail': 'abc@example.com'}
+        }],
+        ['abc@example.com', ],  # output, list of email addresses
+        MESSAGES[0],  # output, expected message fragment
+    ],
```

Comment:
We can probably remove these two cases with `'applicationCompanyDetailsConfirmed': None`  - suppliers cannot complete (or even start!) their declaration until they have confirmed their company details.